### PR TITLE
Add countl_zero function

### PR DESCRIPTION
**Pull request by phprus**
Fix for issue #3213

cc @Epixu @vitaut

### CODE REVIEW

File: test/format-impl-test.cc [hunk 1]
```diff
@@ -353,6 +353,19 @@ TEST(format_impl_test, count_digits) {
...
 }
 
+// Tests fmt::detail::countl_zero for unsigned type UInt.
+template <typename UInt> void test_countl_zero() {
+  constexpr auto num_bits = fmt::detail::num_bits<UInt>();
+  UInt n = 1u;
+  for (int i = 1; i < num_bits - 1; i++) {
+    n <<= 1;
+    EXPECT_EQ(fmt::detail::countl_zero(n - 1), (num_bits - i));
+    EXPECT_EQ(fmt::detail::countl_zero(n), (num_bits - i - 1));
```
[the remark this answers is not in the document]
Comment (phprus) [reply test/format-impl-test.cc]: Done.
And minor change in ``countl_zero``.

File: test/format-impl-test.cc [hunk 2]
```diff
@@ -353,6 +353,19 @@ TEST(format_impl_test, count_digits) {
...
+// Tests fmt::detail::countl_zero for unsigned type UInt.
+template <typename UInt> void test_countl_zero() {
+  constexpr auto num_bits = fmt::detail::num_bits<UInt>();
+  UInt n = 1u;
+  for (int i = 1; i < num_bits - 1; i++) {
+    n <<= 1;
+    EXPECT_EQ(fmt::detail::countl_zero(n - 1), num_bits - i);
+    EXPECT_EQ(fmt::detail::countl_zero(n), num_bits - i - 1);
+  }
+}
```
Comment (vitaut) [inline test/format-impl-test.cc:365]: Do we need this template? Why not merge the logic into the test and get rid of it?

Comment (phprus) [reply test/format-impl-test.cc]: Merged.

**Comment by vitaut** [comment]
Thank you!

**Pull request merged**